python/lib/get_session_info.py

```diff
@@ -0,0 +1,328 @@
+def get_dicom_archive_session_info(env: Env, dicom_archive: DbDicomArchive) -> SessionInfo:
+    """
+    Get the session information for a DICOM archive database object using the session
+    identification configuration function, or raise a `SessionConfigError` if the configuration
+    returned is incorrect.
+    """
+
+    patient_id_dicom_header = get_patient_id_dicom_header_config(env)
+    match patient_id_dicom_header:
+        case 'PatientID':
+            patient_id = dicom_archive.patient_id
+        case 'PatientName':
+            patient_id = dicom_archive.patient_name
+
+    scanner_info = get_dicom_archive_scanner_info(dicom_archive)
+
+    return get_session_info(env, patient_id, scanner_info)
+
+
+def get_session_info(env: Env, patient_id: str, scanner_info: MriScannerInfo) -> SessionInfo:
+    """
+    Get the session information for a patient ID using the session identification configuration
+    function, or raise a `SessionConfigError` if the configuration returned is incorrect.
+    """
+
+    try:
+        session_config = env.config_info.get_session_config(env.db, patient_id)
+    except AttributeError:
+        raise SessionConfigError("Function `get_session_config` not found in the Python configuration file.")
+
+    if session_config is None:
+        raise SessionConfigError(
+            f"No session configuration returned by `get_session_config` for patient ID '{patient_id}'."
+        )
+
+    return get_session_config_info(env, session_config, scanner_info)
+
+
+def get_session_config_info(env: Env, session_config: SessionConfig, scanner_info: MriScannerInfo) -> SessionInfo:
+    """
+    Get the session information for a session configuration, or raise a `SessionConfigError` if
+    that configuration is incorrect.
+    """
+
+    match session_config:
+        case SessionCandidateConfig():
+            return get_candidate_session_info(env, session_config, scanner_info)
+        case SessionPhantomConfig():
+            return get_phantom_session_info(env, session_config, scanner_info)
+
+
+def get_candidate_session_info(
+    env: Env,
+    session_config: SessionCandidateConfig,
+    scanner_info: MriScannerInfo,
+) -> SessionInfo:
+    """
+    Get the session information for a candidate session configuratution, or raise a
+    `SessionConfigError` if that configuration is incorrect.
+    """
+
+    candidate = try_get_candidate_with_cand_id(env.db, session_config.cand_id)
+
+    if candidate is None:
+        raise_session_config_error(
+            session_config,
+            f"No candidate found for CandID {session_config.cand_id}."
+        )
+
+    if candidate.psc_id != session_config.psc_id:
+        raise_session_config_error(
+            session_config,
+            (
+                f"Candidate {session_config.cand_id} PSCID does not match the session configuration PSCID.\n"
+                f"The candidate PSCID is '{candidate.psc_id}', the session configuration PSCID is"
```

Comment:
instead of talking about the session configuration PSCID, maybe simplify by saying that the PSCID stored in the DICOM header does not match the CandID stored in the DICOM header. More intuitive for users not knowing the code in depth.